src/js/utils/ActionDots.jsx

```diff
@@ -5,39 +5,72 @@ import { RiMoreLine } from 'react-icons/all';
-    </div>
-    <div className="dropdown-menu dropdown-menu-right nav-item padding-8">
-      {actions && actions.map(action => (
-        <React.Fragment key={action.href ? action.href : action.label}>
-          {action.href && (
-          <a key={action.href} href={`${action.href}/${id}`} className={`d-flex align-items-center gap-8 dropdown-item ${action.variant === 'danger' ? 'font-red-ob' : ''}`}>
-            {action.leftIcon && action.leftIcon}
-            {action.label && <Translate id="react.purchaseOrder.viewOrderDetails.label" defaultMessage={action.defaultLabel} />}
-          </a>
-          )}
-          {action.onClickActionWithId && (
-          <a key={action.label} href="#" onClick={() => action.onClickActionWithId(id)} className={`d-flex align-items-center gap-8 dropdown-item ${action.variant === 'danger' ? 'font-red-ob' : ''}`}>
-            {action.leftIcon && action.leftIcon}
-            {action.label && <Translate id="react.purchaseOrder.viewOrderDetails.label" defaultMessage={action.defaultLabel} />}
-          </a>
-          )}
-        </React.Fragment>
-      ))}
+const ActionDots = ({
+  actions, id, dropdownPlacement, dropdownClasses,
+}) => {
+  const getPositionClass = () => {
+    switch (dropdownPlacement) {
+      case 'top':
+        return 'dropup';
+      case 'left':
+        return 'dropleft';
+      case 'right':
+        return 'dropright';
+      default:
+        return '';
+    }
+  };
+
+  return (
+    <div className={`btn-group ${getPositionClass()}`}>
+      <div
+        className="action-dots dropdown-toggle d-flex align-items-center justify-content-center"
+        data-toggle="dropdown"
+        aria-haspopup="true"
+        aria-expanded="false"
+      >
+        <RiMoreLine />
+      </div>
+      <div className={`${dropdownClasses} dropdown-menu dropdown-menu-right nav-item padding-8`}>
+        {actions && actions.map(action => (
+          <React.Fragment key={action.href ? action.href : action.label}>
+            {action.href && (
+              <a
+                key={action.href}
+                href={`${action.href}/${id}`}
+                className={`d-flex align-items-center gap-8 dropdown-item ${action.variant === 'danger' ? 'font-red-ob' : ''}`}
+              >
+                {action.leftIcon && action.leftIcon}
+                {action.label &&
+                <Translate id={action.label} defaultMessage={action.defaultLabel} />}
+              </a>
+            )}
+            {action.onClickActionWithId && (
+              <a
+                key={action.label}
+                href="#"
+                onClick={() => action.onClickActionWithId(id)}
+                className={`d-flex align-items-center gap-8 dropdown-item ${action.variant === 'danger' ? 'font-red-ob' : ''}`}
+              >
+                {action.leftIcon && action.leftIcon}
+                {action.label &&
+                <Translate id={action.label} defaultMessage={action.defaultLabel} />}
```

Comment:
nice catch, I haven't realized I hardcoded id of translation as `react.purchaseOrder.viewOrderDetails.label` (translations weren't working for me, so I got defaultMessage always instead)